cpp/src/io/orc/orc.cpp

```diff
@@ -471,20 +469,11 @@ void metadata::init_column_names() const
   auto const &types      = ff.types;
   for (int32_t col_id = 0; col_id < get_num_columns(); ++col_id) {
     std::string col_name;
-    uint32_t parent_idx = col_id;
-    uint32_t idx        = col_id;
-    do {
-      idx        = parent_idx;
-      parent_idx = (idx < types.size()) ? static_cast<uint32_t>(schema_idxs[idx].parent) : ~0;
-      if (parent_idx >= types.size()) break;
-
-      auto const field_idx =
-        (parent_idx < types.size()) ? static_cast<uint32_t>(schema_idxs[idx].field) : ~0;
-      if (field_idx < types[parent_idx].fieldNames.size()) {
-        col_name =
-          types[parent_idx].fieldNames[field_idx] + (col_name.empty() ? "" : ("." + col_name));
-      }
-    } while (parent_idx != idx);
+    uint32_t parent_idx = static_cast<uint32_t>(schema_idxs[col_id].parent);
+    uint32_t field_idx  = static_cast<uint32_t>(schema_idxs[col_id].field);
```

Comment:
The if condition in the next line will take care of that scenario.

Reply:
Not a big deal, but the logic here is implicit/fragile. The invalid value (`-1`) is only covered in the next line because of unsigned integer underflow. 
IMO there should be a validity check for `schema_idxs[col_id].field` (and maybe `schema_idxs[col_id].parent`, not sure) before we compare against `fieldNames.size()` and potentially set the column name.